src/irreps/anirrep.jl

```diff
@@ -0,0 +1,304 @@
+"""
+    struct ANIrrep{N} <: AbstractIrrep{Alternating{N}}
+    ANIrrep{N}(n::Integer)
+    Irrep[Alternating{N}](n::Integer)
+
+Represents irreps of the alternating group ``A_N``.
+
+## Fields
+
+- `n::Int`: the label of the irrep.
+
+This can take the value 
+"""
+struct ANIrrep{N} <: AbstractIrrep{Alternating{N}}
+    n::Int
+    function ANIrrep{N}(n::Int) where {N}
+        @assert 0 < N < 5 "ANIrrep is only provided for 0 < N < 5"
+        0 <= n <= _npartitions(N) - 1 ||
+        throw(DomainError(n, "ANIrrep only has irreps " * lazy"0 <= n <= $(_npartitions(N) - 1)"))
+        return new{N}(n)
+    end
+end
+
+_npartitions(N::Int) = (N == 3) ? 3 : (N == 4) ? 4 : 1
+
+Base.propertynames(x::ANIrrep) = (:n,)
+
+function Base.getproperty(a::ANIrrep{N}, sym::Symbol) where {N}
+    if sym === :n
+        return getfield(a, :n)
+    else
+        error("Unknown property $sym")
+    end
+end
+
```

Comment:
```suggestion
```

This is the default implementation for `getproprerty`, so you don't need to overload this :)